sklearn/metrics/_classification.py

```diff
@@ -360,6 +376,24 @@ def accuracy_score(y_true, y_pred, *, normalize=True, sample_weight=None):
     y_type, y_true, y_pred = _check_targets(y_true, y_pred)
     check_consistent_length(y_true, y_pred, sample_weight)
 
+    if _num_samples(y_true) == 0:
+        if normalize:
+            msg = (
+                "`y_true` and `y_pred` are empty. `accuracy_score` is undefined and "
+                "set to the value defined in the `replace_undefined_by` param, which "
+                "defaults to 0.0."
+            )
+            warnings.warn(msg, UndefinedMetricWarning, stacklevel=2)
+            return replace_undefined_by
+        else:
+            msg = (
+                "`y_true` and `y_pred` are empty. `accuracy_score` is undefined and "
+                "set to the value defined in the `replace_undefined_by` param, which "
+                "defaults to 0 when `normalize=False` is set."
+            )
```

Comment:
looks like we can use the same warning message, except the last part, which we can remove, they're the same.